File: android/src/ti/airship/events/NotificationResponseEvent.java
/* Copyright Airship and Contributors */

package ti.airship.events;

import androidx.annotation.NonNull;

import java.util.Map;

import ti.airship.TIAirshipModule;

public class NotificationResponseEvent implements Event {
    private final TiNotificationResponse response;

    public NotificationResponseEvent(@NonNull TiNotificationResponse response) {
        this.response = response;
    }

    @NonNull
    @Override
    public Map<String, Object> getData() {
        return response.toMap();
    }

    @NonNull
    @Override
    public String getName() {
        return TIAirshipModule.EVENT_NOTIFICATION_RESPONSE;
    }
}

File: android/src/ti/airship/events/NotificationResponseEvent.kt
/* Copyright Airship and Contributors */

package ti.airship.events

import com.urbanairship.push.NotificationActionButtonInfo
import com.urbanairship.push.NotificationInfo
import ti.airship.TiAirshipModule

class NotificationResponseEvent(notificationInfo: NotificationInfo, buttonInfo: NotificationActionButtonInfo?) : Event {
    override val data: Map<String, Any> = notificationInfo.eventPayload(buttonInfo)
    override val name: String = TiAirshipModule.eventNotificationResponseReceived
}